applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_incremental_linear_elastic_interface_law.cpp

```diff
@@ -31,100 +30,149 @@
 using namespace Kratos;
 using namespace std::string_literals;
 
-namespace Kratos::Testing
+namespace
+{
+GeoIncrementalLinearElasticInterfaceLaw CreateLaw2D()
 {
+    return GeoIncrementalLinearElasticInterfaceLaw{std::make_unique<InterfacePlaneStrain>()};
+}
 
-KRATOS_TEST_CASE_IN_SUITE(LinearElasticLawForInterfacesHas2DWorkingSpace, KratosGeoMechanicsFastSuiteWithoutKernel)
+GeoIncrementalLinearElasticInterfaceLaw CreateLaw3D()
 {
-    auto law = GeoIncrementalLinearElasticInterfaceLaw{std::make_unique<InterfacePlaneStrain>()};
+    return GeoIncrementalLinearElasticInterfaceLaw{std::make_unique<InterfaceThreeDimensionalSurface>()};
+}
 
-    KRATOS_EXPECT_EQ(law.WorkingSpaceDimension(), 2);
+void InitializeLawMaterial(GeoIncrementalLinearElasticInterfaceLaw& rLaw,
+                           const Properties&                        rProperties = Properties{})
+{
+    const auto dummy_geometry              = Geometry<Node>{};
+    const auto dummy_shape_function_values = Vector{};
+    rLaw.InitializeMaterial(rProperties, dummy_geometry, dummy_shape_function_values);
 }
 
-KRATOS_TEST_CASE_IN_SUITE(LinearElasticLawForInterfacesHas3DWorkingSpace, KratosGeoMechanicsFastSuiteWithoutKernel)
+void SetLawInitialState(GeoIncrementalLinearElasticInterfaceLaw& rLaw,
+                        const Vector&                            rInitialRelativeDisplacement,
+                        const Vector&                            rInitialTraction)
 {
-    auto law = GeoIncrementalLinearElasticInterfaceLaw{std::make_unique<InterfaceThreeDimensionalSurface>()};
+    const auto p_initial_state = make_intrusive<InitialState>(rInitialRelativeDisplacement, rInitialTraction);
+    rLaw.SetInitialState(p_initial_state);
+}
 
-    KRATOS_EXPECT_EQ(law.WorkingSpaceDimension(), 3);
+void TestInitialStates(GeoIncrementalLinearElasticInterfaceLaw& rLaw,
+                       const Vector&                            rInitialRelativeDisplacement,
+                       const Vector&                            rInitialTraction)
```

Comment:
Perhaps we can add "Expected" to the function parameter names to make clearer what it means:
```suggestion
                       const Vector&                            rExpectedInitialRelativeDisplacement,
                       const Vector&                            rExpectedInitialTraction)
```

Reply:
changed